tests/link/test_numba_performance.py

```diff
@@ -0,0 +1,80 @@
+import time
+
+import numpy as np
+import pytest
+
+import aesara.tensor as aet
+from aesara import config
+from aesara.compile.function import function
+from aesara.compile.mode import Mode
+from aesara.compile.sharedvalue import SharedVariable
+from aesara.graph.basic import Constant
+from aesara.graph.fg import FunctionGraph
+from aesara.graph.optdb import OptimizationQuery
+from aesara.link.numba.linker import NumbaLinker
+
+
+opts = OptimizationQuery(include=[None], exclude=["cxx_only", "BlasOpt"])
+numba_mode = Mode(NumbaLinker(), opts)
+py_mode = Mode("py", opts)
+
+
+def set_test_value(x, v):
+    x.tag.test_value = v
+    return x
+
+
+def run_numpy_benchmark(fgraph, numpy_fn, inputs, numpy_inputs, performace_factor=1):
+    fn_inputs = [i for i in fgraph.inputs if not isinstance(i, SharedVariable)]
+    aesara_numba_fn = function(
+        fn_inputs,
+        fgraph.outputs,
+        mode=numba_mode,
+        accept_inplace=True,
+    )
+    numba_res = aesara_numba_fn(*inputs)
+    numpy_res = numpy_fn(*numpy_inputs)
+
+    assert np.all(numba_res == numpy_res)
+
+    t0 = time.time()
+    aesara_numba_fn(*inputs)
+    t1 = time.time()
+    numpy_fn(*numpy_inputs)
+    t2 = time.time()
+
+    assert t1 - t0 <= performace_factor * (t2 - t1)
```

Comment:
We need to use `timeit` for these kinds of comparisons.  

(Also, see [`time.perf_counter`](https://docs.python.org/3/library/time.html#time.perf_counter) as a slightly better alternative to `time.time`.)